stl/inc/vector

```diff
@@ -1031,32 +1044,15 @@ private:
         pointer& _Myend     = _My_data._Myend;
 
         _My_data._Orphan_all();
-
-        if constexpr (conjunction_v<bool_constant<_Ptr_copy_cat<_Iter, _Ty*>::_Trivially_copyable>,
-                          _Uses_default_construct<_Alty, _Ty*, decltype(*_First)>,
-                          _Uses_default_destroy<_Alty, _Ty*>>) {
-#if _HAS_CXX20
-            if (!_STD is_constant_evaluated())
-#endif // _HAS_CXX20
-            {
-                const auto _Oldcapacity = static_cast<size_type>(_Myend - _Myfirst);
-                if (_Newsize > _Oldcapacity) {
-                    _Clear_and_reserve_geometric(_Newsize);
-                }
-
-                _Mylast = _Refancy<pointer>(_Copy_memmove(_First, _Last, _Unfancy(_Myfirst)));
-                return;
-            }
+        const auto _Oldcapacity = static_cast<size_type>(_Myend - _Myfirst);
+        if (_Newsize > _Oldcapacity) {
+            _Clear_and_reserve_geometric(_Newsize);
+            _Mylast = _Uninitialized_copy(_First, _Last, _My_data._Myfirst, _Al);

```

Comment:
I think we can have the optimization _and_ avoid duplicating the reallocation code a la:
```c++
_My_data._Orphan_all();
auto _Oldsize = static_cast<size_type>(_Mylast - _Myfirst);
const auto _Oldcapacity = static_cast<size_type>(_Myend - _Myfirst);
if (_Newsize > _Oldcapacity) {
    _Clear_and_reserve_geometric(_Newsize);
    _Oldsize = 0;
}

if constexpr (conjunction_v<bool_constant<_Ptr_copy_cat<_Iter, _Ty*>::_Trivially_copyable>,
    _Uses_default_construct<_Alty, _Ty*, decltype(*_First)>,
    _Uses_default_destroy<_Alty, _Ty*>>) {
#if _HAS_CXX20
    if (!_STD is_constant_evaluated())
#endif // _HAS_CXX20
    {
        _Mylast = _Refancy<pointer>(_Copy_memmove(_First, _Last, _Unfancy(_Myfirst)));
        return;
    }
}

if (_Newsize > _Oldsize) {
```

Reply:
We do not need it here, because `_Uninitialized_copy` internally applies the optimization. Same for the others

Reply:
Urgh, actually this is not correct if we do not use default destruction

Reply:
> We do not need it here, because `_Uninitialized_copy` internally applies the optimization, see

When I hit the second occurrence, I started to think this was maybe the case, but decided to be lazy and let you tell me =)

Reply:
😲